client/src/components/Anonymous.jsx

```diff
@@ -170,6 +173,61 @@ const Anonymous = ({
 		});
 	};
 
+	const reportUser = async () => {
+		// Get the other user id 
+		const otherUserId = state[currentChatId]?.userIds.find(
+			id => id !== authState.loginId && id !== authState.email
+		  );
+
+		  if (!otherUserId) {
+			return { success: false, message: "Could not identify the user to report" };
+		  }
+
+	 	 try {
+			const res = await api.post('/reportUser', {
+			  userIdToReport: otherUserId,
+			  currentUserId: authState.loginId
+			});
+		
+			if (res.status === 200) {
+			  return { success: true };
+			} else {
+			  return { success: false, message: "Error reporting user" };
+			}
+		  } catch (error) {
+			console.error("Error in reportUser:", error);
+			return { success: false, message: "An unexpected error occurred" };
+		  }
+	}
+
+	const handleReport = async () => {
+		// Check if user have an account i.e. not a anonymous user
+		if(authState.loginType === "anonymous") {
+			 setDialog({
+				isOpen: true,
+				text: "You have to create an account first to access this feature!",
+				yesBtnText: "Create an account",
+				noBtnText: "Back to chat",
+				handler: () => navigate("/profile")
+			 })
+			 return
+		}
+
+		try {
+			const result = await reportUser();
+			if (result.success) {
+			  alert('User reported and blocked successfully');
+			  // Update the UI or redirect the user here
+			  closeChatHandler(false)
+			} else {
+			  alert(result.message || "Error reporting user. Please try again later.");
+			}
+		  } catch (err) {
+			console.error("Error in handleReport:", err);
+			alert("User is already blocked!");
```

Comment:
This is not correct because we aren't checking if the user was actually being blocked, so this is a wrong message and unless you checked if the user was already blocked then it's not a good UX